modules/querying/pages/func/query-user-defined-functions.adoc

```diff
@@ -75,22 +66,42 @@ inline string reverse(string str){
 }
 ----
 
-== Use GitHub to store UDFs
+== Upload UDFs
+
+There are three ways to modify these files to add user-defined functions to GSQL:
+
+* Store UDF files in a GitHub repository, and configure GSQL to read from the repository.
+** This is the recommended approach.
+It also takes the highest precedence.
+If GSQL is configured to read from GitHub for UDFs, and the configurations are valid, local UDF files are ignored.
+* Use `gadmin` commands to upload a local UDF file to GSQL.
+** Use this approach when your server does not have access to the internet and cannot access GitHub.
+* Use `GET` and `PUT` commands to download, and upload the files directly through GSQL shell.
+** For security reasons, this method is disabled by default.
```

Comment:
@mingxiw @qianzhang-tg For the official documentation let's make sure we get the exact wording that we want here. How we word this is important.

I could see us phrase the new gadmin config, and github, as secure udf file offerings, omitting PUT as a less favorable option. 

We can add a footnote or * or something to select github and gadmin config and indicate these offerings provide the most secure and convenient way to manage UDFs. Then have a disclaimer on the PUT option that this is disabled by default and should only be used sparingly, with deliberation.

What are your thoughts?

Reply:
I agree that the documentation should add some visual cues to show which one is desired and which one is not.

Reply:
I can try to take more of a pros and cons approach to this - saying that using PUT gives more flexibility as you don't need to have access to the file systems of the server, but it might also introduce security risks, unless we want to be very explicit about its flaws.

Reply:
Hi Lenny - I recommend focusing on GitHub and gadmin config offering rather than PUT. I don't want us to emphasize that PUT creates security vulnerabilities. We don't want to accept Neo4j's claims. Instead, we will focus on the GitHub and gadmin config as the recommended secured offering & how it is more secure for users/customers. That should be our approach. I can review it once it is finalized. Thanks Lenny and team.